### PR TITLE
guava-bom

### LINKED ISSUE #505: Add guava-bom to cloud-oss-bom

### PR DESCRIPTION
**Pull request by elharo**
fix #505 @garrettjonesgoogle

### CODE REVIEW

File: dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/ClassPathBuilderTest.java [hunk 1]
```diff
@@ -77,17 +77,21 @@ public void testArtifactsToPaths_removingDuplicates() throws RepositoryException
 
   @Test
   public void testBomToPaths_firstElementsAreBomMembers() throws RepositoryException {
+    
+    // that is, that the BOM members come before the transitive dependencies
```
Comment (garrettjonesgoogle) [inline dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/ClassPathBuilderTest.java]: This comment just seems to be floating and incomplete

Comment (elharo) [reply dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/ClassPathBuilderTest.java]: It follows the method name.

Comment (garrettjonesgoogle) [reply dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/ClassPathBuilderTest.java]: Statements that cross modalities can be very confusing. I'd much prefer a self-contained statement.

Comment (elharo) [reply dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/ClassPathBuilderTest.java]: Moved the comment. This is a weird test. I'm not 100% sure it's correct or that we need it. By Chesterton's fence I left it in, but it's worth another look.

File: dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/ClassPathBuilderTest.java [hunk 2]
```diff
@@ -75,8 +75,11 @@ public void testArtifactsToPaths_removingDuplicates() throws RepositoryException
         .isGreaterThan(1);
   }
 
+  /**
+   * Test that BOM members come before the transitive dependencies.
+   */
   @Test
-  public void testBomToPaths_firstElementsAreBomMembers() throws RepositoryException {
+  public void testBomToPaths_firstElementsAreBomMembers() throws RepositoryException {
```
Comment (garrettjonesgoogle) [inline dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/ClassPathBuilderTest.java]: remove trailing whitespace

File: boms/cloud-oss-bom/pom.xml [hunk 3]
```diff
@@ -29,7 +29,7 @@
 
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <guava.version>26.0-android</guava.version>
+    <guava.version>27.1-android</guava.version>
```
Comment (netdpb) [inline boms/cloud-oss-bom/pom.xml]: Now that this property is used only once, can we inline it instead of using a property?

Comment (elharo) [reply boms/cloud-oss-bom/pom.xml]: We can. Personally I like having all the versions in one place though. Anyone else have a strong opinion about this?

Comment (netdpb) [reply boms/cloud-oss-bom/pom.xml]: I thought that single place could be the `<dependencyManagement>` section.

Comment (elharo) [reply boms/cloud-oss-bom/pom.xml]: We don't have BOMs for everything yet, and even when we do they're still going to be separated form each other by multiple lines.

File: dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/ClassPathBuilderTest.java (hunk 2 above)
Comment (elharo) [reply dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/ClassPathBuilderTest.java]: https://engdoc.corp.google.com/eng/doc/devguide/java/style/index.md?cl=head#s8.1.1-no-comment-issues

File: boms/cloud-oss-bom/pom.xml (hunk 3 above)
Comment (netdpb) [reply boms/cloud-oss-bom/pom.xml]: This is up to you all.

**Comment by netdpb** [comment]
If we're using google-java-format, this stuff should go away anyway.

**Pull request merged**